### PR TITLE
fix: use shell-compatible curl install

### LINKED ISSUE #3664: Change Linux Installation Command for Compatiblity
#### Is your feature request related to a problem? Please describe.

Installing starship in a non-POSIX compliant shell (such as Fish) can cause some confusion when the installation can't be executed.
Most non-compliant shells tend to have a special syntax for command substitution, which is why it's better to have a general form of the install command.

#### Describe the solution you'd like

By changing the installation command for Linux to:
`curl -fsSL https://starship.rs/install.sh | sh`
it allows starship to be installed in non-compliant shells while keeping the same behavior for complaint shells. Piping the curl command to sh is also seen in [rustup.rs](https://rustup.rs)
#### Describe alternatives you've considered

It is easy to say that the installation script should run initially in BASH before being usable in non-compliant shells, but that can be inconvenient for the end user.

### PR DESCRIPTION
**Pull request by matchai**
#### Description

Update curl installation method to not use command expansion.

#### Motivation and Context

Closes #3664

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,7 +203,7 @@ Install Starship using any of the following package managers:
 Install the latest version for your system:
 
 ```sh
-sh -c "$(curl -fsSL https://starship.rs/install.sh)"
+curl -sS https://starship.rs/install.sh | sh
```
Comment (davidkna) [inline README.md:206]: You also need to update L235 of this file, `docs/README.md`, `docs/installing/README.md` and `docs/faq/README.md`

**Review by chipbuster — APPROVED** [review]
:+1: LGTM

**Comment by thecesrom** [comment]
I am installing starship on Git Bash for Windows, and this command produces the following otuput.

New:

```bash
$ curl -sS https://starship.rs/install.sh | sh -s -- --bin-dir "$HOME/.starship/bin"

  Configuration
> Bin directory: /c/Users/thecesrom/.starship/bin
> Platform:      pc-windows-msvc
> Arch:          x86_64

> Tarball URL: https://github.com/starship/starship/releases/latest/download/starship-x86_64-pc-windows-msvc.zip
? Install Starship latest to /c/Users/thecesrom/.starship/bin? [y/N] y
> Installing Starship, please wait…
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [completed]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [ "Starshi]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [p install]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [ed"]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [{ENTER}]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: error:  invalid response [printf '\]
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename:
$ echo $?
0
```

Apparently the installation was completed (see the exit code `0`), but it does not make for a good user experience.

Using the "old" command, as expected, runs unaffected.

```bash
$ sh -c "$(curl -fsSL https://starship.rs/install.sh)" -- --bin-dir "$HOME/.starship/bin"

  Configuration
> Bin directory: /c/Users/thecesrom/.starship/bin
> Platform:      pc-windows-msvc
> Arch:          x86_64

> Tarball URL: https://github.com/starship/starship/releases/latest/download/starship-x86_64-pc-windows-msvc.zip
? Install Starship latest to /c/Users/thecesrom/.starship/bin? [y/N] y
> Installing Starship, please wait…
replace /c/Users/thecesrom/.starship/bin/starship.exe? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
✓ Starship installed
```

On the other hand, when passing `--help` as an option 

Thanks!

**Comment by chipbuster** [comment]
@thecesrom Thanks for bringing this up! I wasn't able to trivially reproduce it on Linux (I suspect that the `replace <file>? [y]es, [n]o, [A]ll, [N]one, [r]ename:` output is due to something specific to Windows) so I've gone ahead and opened a new issue for it #3726

It turns out I goobered the 1.4.1 release due to inexperience, so I'm hoping we can get this, #3723, and #3705 sorted out before I make a stab at the 1.4.2 release later today.

**Pull request merged**